### PR TITLE
Fix event subscription and eFormidling status check.

### PR DESCRIPTION
**Pull request by RonnyB71**
This change checks to see if an implementation for IEventSubscription is already added to the service collection before adding it.
Add more logging to the EventSubscriptionClient and other areas.

## Related Issue(s)
- #49 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.App.Core/Internal/Events/EventHandlerResolver.cs [hunk 1]
```diff
@@ -16,6 +16,11 @@ public EventHandlerResolver(IEnumerable<IEventHandler> eventHandlers)
         /// <inheritDoc/>
```
Comment (tjololo) [inline src/Altinn.App.Core/Internal/Events/EventHandlerResolver.cs]:
```suggestion
        /// <inheritdoc/>
```

File: src/Altinn.App.Api/Controllers/EventsReceiverController.cs [hunk 2]
```diff
@@ -62,11 +69,11 @@ public async Task<ActionResult> Post([FromQuery] string code, [FromBody] CloudEv
             }
             catch (NotImplementedException)
             {
-                return BadRequest();
+                return BadRequest($"No eventhandler found that supports {cloudEvent.Type}");
             }
             catch (Exception ex)
```
Comment (tjololo) [inline src/Altinn.App.Api/Controllers/EventsReceiverController.cs:74]: Catching and handling all exceptions in controllers makes sense to me, should we dismiss the CodeQL alert with won't fix?

**Review by tjololo — APPROVED** [review]
:godmode:

File: src/Altinn.App.Api/Controllers/EventsReceiverController.cs (hunk 2 above)
Comment (RonnyB71) [reply src/Altinn.App.Api/Controllers/EventsReceiverController.cs:74]: Yes, in this particular case I wanted a more explicit logging that we would get with the generic error handling which also would return a 500 but with less info logged.

**Pull request merged**